crates/red_knot_python_semantic/resources/mdtest/call/dunder.md

```diff
@@ -0,0 +1,125 @@
+# Dunder calls
+
+## Introduction
+
+This test suite explains and documents how dunder methods are looked up and called. Throughout the
+document, we use `__getitem__` as an example, but the same principles apply to other dunder methods.
+
+Dunder methods are implicitly called when using certain syntax. For example, the index operator
+`obj[key]` calls the `__getitem__` method under the hood. Exactly *how* a dunder method is looked up
+and called works slightly different from regular methods. Dunder methods are not looked up on `obj`
+directly, but rather on `type(obj)`. But in many ways, they still *act* as if they were called on
+`obj` directly. If the `__getitem__` member of `type(obj)` is a descriptor, it is called with `obj`
+as the `instance` argument to `__get__`. A desugared version of `obj[key]` is roughly equivalent to
+`getitem_desugared(obj, key)` as defined below:
+
+```py
+from typing import Any
+
+def find_in_mro(typ: type, name: str) -> Any: ...
```

Comment:
readers of the test may not be familiar with this method. It's possibly not worth it to fill in the body here since it's not very relevant to the test, but you could maybe link to Brett's blog or the Descriptor HOWTO docs in the docstring for this function if readers want to find out how it works?

Though the function isn't that long IIRC; it might be easier just to define it here.

Reply:
I feel like regardless of familiarity, the name is sufficient here to explain what it does, and implementing it might be a distraction from the main point here, which is about descriptor behavior, not how names are looked up in base classes.

Reply:
> I feel like regardless of familiarity, the name is sufficient here to explain what it does

Hmm, I sort-of disagree. I do agree that the implementation of the method isn't the main point of the test, but I think an inline link to the Descriptor HOWTO (for example) wouldn't really distract readers

Reply:
> I think an inline link to the Descriptor HOWTO (for example), which provides an implementation of this function, wouldn't really distract readers

Feels like a good compromise to me. I might also change that example here again when working on https://github.com/astral-sh/ruff/issues/16367